python/GafferUITest/BackgroundMethodTest.py

```diff
@@ -107,15 +107,15 @@ def __setTextPlug( self ) :
 
 			return self.__script["n"]["sum"]
 
-	class WaitingSlot( GafferTest.CapturingSlot ) :
+	class AwaitSetValueSlot( GafferTest.CapturingSlot ) :
 
-		def __init__( self, signal ) :
+		def __init__( self, valueWidget ) :
 
-			GafferTest.CapturingSlot.__init__( self, signal )
+			GafferTest.CapturingSlot.__init__( self, valueWidget.valueChangedSignal() )
 
 		def wait( self ) :
 
-			while len( self ) == 0 :
+			while not [ e for e in self if e[1] == GafferUI.NumericWidget.ValueChangedReason.SetValue ] :
```

Comment:
I'd like to know more about why `setEnabled()` causes `valueChangedSignal()` to be emitted - that seems wrong. Has that always been the case or is it new to Qt 5.12? Only on Mac? I think there may be a decent case for fixing this at source rather than here.

Reply:
It does this on Linux too, if you `self.waitForIdle` long enough. I think it's maybe expected behaviour. The text field has keyboard focus by default, disabling it emits `editingFinished`, which consequently causes our `valueChangedSignal` to do its thing.

```
  File "/data/dev/gaffer-build/python/GafferUITest/BackgroundMethodTest.py", line 93, in __updateInBackgroundPreCall
    self.setEnabled( False )
  File "/data/dev/gaffer-build/python/GafferUI/Widget.py", line 282, in setEnabled
    self.__qtWidget.setEnabled( enabled )
  File "/data/dev/gaffer-build/python/GafferUI/TextWidget.py", line 414, in event
    return QtWidgets.QLineEdit.event( self, event )
  File "/data/dev/gaffer-build/python/Gaffer/WeakMethod.py", line 65, in __call__
    return self.__method( s, *args, **kwArgs )
  File "/data/dev/gaffer-build/python/GafferUI/TextWidget.py", line 284, in __editingFinished
    self.__editingFinishedSignal( self )
  File "/data/dev/gaffer-build/python/Gaffer/WeakMethod.py", line 65, in __call__
    return self.__method( s, *args, **kwArgs )
  File "/data/dev/gaffer-build/python/GafferUI/NumericWidget.py", line 273, in __editingFinished
    self.__emitValueChanged( self.ValueChangedReason.Edit )
  File "/data/dev/gaffer-build/python/GafferUI/NumericWidget.py", line 305, in __emitValueChanged
    signal( self, reason )
```

Reply:
This behaviour is present in deps `1.6.0` too.

Reply:
Ah, that makes sense. Thanks for the explanation.

Reply:
Cool, happy with the fix just with a better name?